src/api/compatibility/pickAppResource.ts

```diff
@@ -18,6 +18,7 @@ export function createCompatibilityPickAppResource() {
             childItemFilter: convertExpectedChildContextValueToContextValueFilter(options?.expectedChildContextValue)
         });
 
+        context.resourceId = result.id;
```

Comment:
This is needed for client extensions that use this when retrieving a resource via tree item picker (usually the command pallette)